src/PregWrapper.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+class PregWrapper
+{
+    public static function replace($pattern, $replacement, $subject)
+    {
+        // TODO: check if "u" modifier is in the pattern and throw exception if so
```

Comment:
@kubawerlos I would do this check during PHP-CS-Fixer SUT only

Reply:
Instead, I would gracefully add `u` and later gracefully remove `u`, so it wouldn't matter if dev provided `u` in input or not

Reply:
What I meant is that I would dislike having some calls with the flag and some without. All developers should be aware of the `UTF-8` topic and should know that thay have to let the wrapper take care of this, so no `u` flag around the project should be allowed.
IMHO.

Reply:
if you want all dev to be aware, don't differentiate behaviour during SUT

Reply:
What would be the added value of having the check done while I run `php-cs-fixer fix` in my apps?
The Exception would never be thrown because PHP-CS-Fixer would never be released with the *u-flag-thing* not well fixed for all the fixers.
If we check the *u-flag-thing* in all the travis build of PHP-CS-Fixer builds, we know it's everything ok before any release, so we can skip the check during an ordinary fixing run.

Reply:
benefit is maintainability, if sth is protected only by tests it is not enough for me, as tests could not testing it for some weird reason after a year or class will be considered stable and not internal anymore, so any dev using it outside Fixer directly would not have the check performed.

Reply:
Ok then